### ISSUE TITLE
failed to sample program on ubuntu

### ISSUE THREAD

**Issue by charlesxsh** [flamegraph-rs/flamegraph#225]
flamegraph -o xxxxx.svg -- /some/binary

After executed this command, the perf.data is generated but no svg file.

---------------
.... (normal stdout)
[ perf record: Woken up 4 times to write data ]
[ perf record: Captured and wrote 1.242 MB perf.data (75 samples) ]
failed to sample program
---------------

Does anyone know what would be the possible reason for "failed to sample program" or reason for no svg file?

**Comment by djc** [comment]
No. Try searching for this problem with `perf` (which flamegraph is using to generate the profile).

**Comment by Bwahharharrr** [comment]
I'm having the same issue. perf is reporting that my data has been captured without issues..

[ perf record: Woken up 65 times to write data ]
[ perf record: Captured and wrote 17.144 MB perf.data (1081 samples) ]
failed to sample program

**Comment by Bwahharharrr** [comment]
If you are "panic!" in your script then flamegraph won't build the svg

**Comment by smoelius** [comment]
> If you are "panic!" in your script then flamegraph won't build the svg

And similarly if you are using `std::process::exit` (which was my mistake).

**Comment by pozix604** [comment]
> And similarly if you are using `std::process::exit` (which was my mistake).

This should be documented. I ran into this also.

**Comment by djc** [comment]
Happy to take a PR for improved documentation.
